plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewsMetastoreHelper.java

```diff
@@ -0,0 +1,236 @@
+                .setDataColumns(ImmutableList.of(new Column("dummy", HIVE_STRING, Optional.empty())))
+                .setPartitionColumns(ImmutableList.of())
+                .setParameters(createViewProperties(session))
+                .setViewOriginalText(Optional.of(encodeViewData(definition)))
+                .setViewExpandedText(Optional.of(PRESTO_VIEW_EXPANDED_TEXT_MARKER));
+
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(VIEW_STORAGE_FORMAT)
+                .setLocation("");
+        io.trino.plugin.hive.metastore.Table table = tableBuilder.build();
+        PrincipalPrivileges principalPrivileges = isUsingSystemSecurity ? NO_PRIVILEGES : buildInitialPrivilegeSet(session.getUser());
+
+        Optional<io.trino.plugin.hive.metastore.Table> existing = metastore.getTable(schemaViewName.getSchemaName(), schemaViewName.getTableName());
+        if (existing.isPresent()) {
+            if (!replace || !isPrestoView(existing.get())) {
+                throw new ViewAlreadyExistsException(schemaViewName);
+            }
+
+            metastore.replaceTable(schemaViewName.getSchemaName(), schemaViewName.getTableName(), table, principalPrivileges);
+            return;
+        }
+
+        try {
+            metastore.createTable(table, principalPrivileges);
+        }
+        catch (TableAlreadyExistsException e) {
+            throw new ViewAlreadyExistsException(e.getTableName());
+        }
+    }
+
+    public void dropView(SchemaTableName schemaViewName)
+    {
+        if (getView(schemaViewName).isEmpty()) {
+            throw new ViewNotFoundException(schemaViewName);
+        }
+
+        try {
+            metastore.dropTable(schemaViewName.getSchemaName(), schemaViewName.getTableName(), true);
+        }
+        catch (TableNotFoundException e) {
+            throw new ViewNotFoundException(e.getTableName());
+        }
+    }
+
+    public List<SchemaTableName> listViews(Optional<String> database)
+    {
+        return listDatabases(database).stream()
+                .flatMap(this::listViews)
+                .collect(toImmutableList());
+    }
+
+    public Map<SchemaTableName, ConnectorViewDefinition> getViews(Optional<String> schemaName)
+    {
+        ImmutableMap.Builder<SchemaTableName, ConnectorViewDefinition> views = ImmutableMap.builder();
+        for (SchemaTableName name : listViews(schemaName)) {
+            try {
+                getView(name).ifPresent(view -> views.put(name, view));
+            }
+            catch (TrinoException e) {
+                if (e.getErrorCode().equals(TABLE_NOT_FOUND.toErrorCode())) {
+                    // Ignore view that was dropped during query execution (race condition)
+                }
+                else {
+                    throw e;
+                }
+            }
+        }
+        return views.buildOrThrow();
+    }
+
+    private List<String> listDatabases(Optional<String> database)
```

Comment:
nit: ordering of the private methods is not quite ideal.
`listDatabases` is used in `listViews` and should probably appear right after this method.